truco_mobile/lib/src/model/deck_model.dart

```diff
@@ -0,0 +1,39 @@
+import 'card_model.dart';
+
+class Deck {
+  late List<Card> cards;
+
+  Deck() {
+    cards = _generateCards();
+    _shuffle();
+  }
+
+  List<Card> _generateCards() {
+    List<String> ranks = ['4', '5', '6', '7', 'Q', 'J', 'K', 'A', '2', '3'];
+    List<String> suits = ['Ouros', 'Espadas', 'Copas', 'Paus'];
+    List<Card> generatedCards = [];
+
+    for (var suit in suits) {
+      for (var i = 0; i < ranks.length; i++) {
+        generatedCards.add(Card(rank: ranks[i], suit: suit, value: i + 1));
+      }
+    }
+
+    return generatedCards;
+  }
```

Comment:
Seria ideal retirar esse laço do for dentro de um for

Por exemplo:

```suggestion

  List<Card> _generateCards() {
  
    List<String> ranks = ['4', '5', '6', '7', 'Q', 'J', 'K', 'A', '2', '3'];
    List<String> suits = ['Ouros', 'Espadas', 'Copas', 'Paus'];
 
    return suits.expand(suit) => ranks.map((rank) => Card(rank: rank, suit: suit)).toList()).toList()
  }
  
```
Aqui nesse exemplo acima você já retornaria direto sem precisar alocar um espaço para um array vazio sendo que ele é apenas usada para retornar nessa função.

Dando um paracer sobre o que os methods expand, map e toList fazem:

* expand basicamente serve como um flat, ele transforma uma matriz em apenas uma lista;
* o map serve pra transformar um elemento do array para o que quiser 
![image](https://github.com/Leordrgs/TrucoMobile/assets/110867639/3d558fe3-df7d-4827-8fbe-fdb7825874bd)
* e os dois toList é para os retornos da função expand e map que retornam um iterable e não uma list

Reply:
Nesse aqui o valor da carta não está sendo gerado, precisamos resolver essa questão e o Array da List é pequeno e vai aumentando conforme incluo valores. Mas as ideias são boas, otimizar o código. Vou ver o que consigo melhorar aqui